sys/include/puf_sram.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_puf_sram SRAM PUF
+ * @ingroup     sys
+ *
+ * @brief       SRAM based physically unclonable function (PUF)
+ *
+ * @{
+ * @file
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+#ifndef PUF_SRAM_H
+#define PUF_SRAM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "hashes.h"
+#include "thread.h"
+
+/**
+ * @brief SRAM length considered for seeding
+ */
+#ifndef SEED_RAM_LEN
+#define SEED_RAM_LEN     (2048 / sizeof(uint32_t))
+#endif
+
+/**
+ * @brief SRAM marker to detect reboot without power-off
+ *
+ * Source: https://www.random.org/bytes/
+ */
+#define PUF_SRAM_MARKER  (0xad3021ff)
+
+/**
+ * @brief Global seed variable, allocated on the stack in puf_sram.c
```

Comment:
Allocated on the stack is incorrect, it is a normal global variable, but allocated outside of the normal .data and .bss sections